ports/esp32/machine_timer.c

```diff
@@ -73,8 +73,13 @@ STATIC esp_err_t check_esp_err(esp_err_t code) {
 }
 
 void machine_timer_deinit_all(void) {
-    while (MP_STATE_PORT(machine_timer_obj_head) != NULL) {
-        machine_timer_disable(MP_STATE_PORT(machine_timer_obj_head));
+    // Disable, deallocate and remove all timers from list
+    machine_timer_obj_t **t = &MP_STATE_PORT(machine_timer_obj_head);
+    while (*t != NULL) {
+        machine_timer_disable(*t);
+        machine_timer_obj_t **next = &(*t)->next;
+        m_del_obj(machine_timer_obj_t, *t);
+        t = next;
     }
```

Comment:
Does this need a `MP_STATE_PORT(machine_timer_obj_head) = NULL` at the end?

Reply:
Good catch, my C is rusty after many years in the FP world. I have commited a small fix.